compiler/test/compilable/test24511.d

```diff
@@ -0,0 +1,25 @@
+// REQUIRED_ARGS: -c -os=windows
```

Comment:
`-c` is implied for `compilable` tests.

Reply:
I know :), this test is experiencing the same failure on macOS as https://github.com/dlang/dmd/pull/16385#pullrequestreview-2010515890 is, which is that the linker's complaining about `_main` being missing.
I've added the `-main` switch to that test, and am waiting to see if that fixes it before trying again with this one

Reply:
`compilable` tests aren't supposed to be linked either. This case here with a .c file to be cross-compiled to Windows is particularly interesting, as it requires invoking a cross-C-preprocessor.

> ld: Undefined symbols:
>   _main, referenced from:
>       <initial-undefines>
> clang: error: linker command failed with exit code 1 (use -v to see invocation)
> Error: C preprocess command clang failed for file compilable/imports/test24511_c.c, exit status 1

So here on Mac that seems to fail badly, the C preprocessor (Apple clang) trying to link instead of preprocessing.

I guess the tests work on non-Windows too, as you apply `extern(Windows)` regardless of the target platform, and I think that's fine (just ignored on non-Windows AFAIK).

Reply:
Thanks for the elaboration. I think I'll just disable the test on macOS: it works on FreeBSD so I wager it'd work on macOS if it got past the preprocessor.